ambry-store/src/main/java/com.github.ambry.store/BlobStoreStats.java

```diff
@@ -803,20 +815,26 @@ private void forwardScan(Offset startOffset, Offset endOffset) throws StoreExcep
             if (indexValue.isFlagSet(IndexValue.Flags.Delete_Index)) {
               // delete record
               IndexValue originalPut;
-              if (indexValue.getOriginalMessageOffset() != IndexValue.UNKNOWN_ORIGINAL_MESSAGE_OFFSET
+              if (indexValue.getOriginalMessageOffset() == indexValue.getOffset().getOffset()) {
+                // delete record with no put record
```

Comment:
could this be only bcoz of compaction, or due to any other reasons too? (like legacy bugs)